src/synology_dsm/api/core/external_usb.py

```diff
@@ -0,0 +1,204 @@
+"""DSM external USB device data."""
+
+from __future__ import annotations
+
+from synology_dsm.api import SynoBaseApi
+from synology_dsm.helpers import SynoFormatHelper
+
+
+class SynoCoreExternalUSB(SynoBaseApi):
+    """Class for external USB storage devices."""
+
+    API_KEY = "SYNO.Core.ExternalDevice.Storage.USB"
+    REQUEST_DATA = {"additional": '["all"]'}
+
+    async def update(self):
+        """Updates external USB storage device data."""
+        raw_data = await self._dsm.post(self.API_KEY, "list", data=self.REQUEST_DATA)
+        if raw_data:
+            self._data = raw_data
+            if raw_data.get("data"):
+                self._data = raw_data["data"]
```

Comment:
we should store the devices as a dict with the device_id as key into `self.data` to avoid every time iterating over a list of devices.